### ISSUE TITLE
[explicit-module-boundary-types] Support well-known symbols in allowedNames

### ISSUE THREAD

**Issue by michkot** [typescript-eslint/typescript-eslint#4566]
eslint-plugin rule explicit-module-boundary-types does not support whitelisting well-known symbols via the allowedNames option. 

The rule implementation supports only Identifier and Literal AST node keys, while well-known symbols use MemberExpressions.

**Actual result**

given
```JSON
{
  "rules": {
      "@typescript-eslint/explicit-module-boundary-types": ["error"],
  }
}
```

following will always be flagged as "ESLint: Missing return type on function.(@typescript-eslint/explicit-module-boundary-types)"
```TS
{	
       get [Symbol.toStringTag]() {
		return "MyTag" as const;
	}
}
```

**Expected result**

given
```JSON
{
  "rules": {
      "@typescript-eslint/explicit-module-boundary-types": ["error", { 
        "allowedNames": ["@@toStringTag"]
    }],
  }
}
```

following will not be flagged as error
```TS
{	
       get [Symbol.toStringTag]() {
		return "MyTag" as const;
	}
}
```

**Comment by bradzacher** [comment]
could you please provide a more complete code sample which exactly reproduces the error and the expected working result?

**Comment by bradzacher** [comment]
With the advent of `isolatedDeclarations` which is natively builtin to TypeScript -- this rule is now feature frozen. As such we are closing feature requests for the rule.
